test/MorphoActions.t.sol

```diff
@@ -0,0 +1,194 @@
+import {IERC4626} from "openzeppelin/interfaces/IERC4626.sol";
+import {IERC20} from "openzeppelin/interfaces/IERC20.sol";
+import {IMorpho, MarketParams, Position} from "src/interfaces/IMorpho.sol";
+import {QuarkWallet} from "quark-core/src/QuarkWallet.sol";
+import {QuarkStateManager} from "quark-core/src/QuarkStateManager.sol";
+
+import {QuarkWalletProxyFactory} from "quark-proxy/src/QuarkWalletProxyFactory.sol";
+
+import {YulHelper} from "./lib/YulHelper.sol";
+import {SignatureHelper} from "./lib/SignatureHelper.sol";
+import {QuarkOperationHelper, ScriptType} from "./lib/QuarkOperationHelper.sol";
+
+import "src/MorphoScripts.sol";
+
+/**
+ * Tests for Morpho Blue market
+ */
+contract MorphoActionsTest is Test {
+    QuarkWalletProxyFactory public factory;
+    uint256 alicePrivateKey = 0xa11ce;
+    address alice = vm.addr(alicePrivateKey);
+
+    // Contracts address on mainnet
+    address constant morpho = 0xBBBBBbbBBb9cC5e90e3b3Af64bdAF62C37EEFFCb;
+    address constant adaptiveCurveIrm = 0x870aC11D48B15DB9a138Cf899d20F13F79Ba00BC;
+    address constant morphoOracle = 0x48F7E36EB6B826B2dF4B2E630B62Cd25e89E40e2;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant wstETH = 0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    MarketParams marketParams = MarketParams(USDC, wstETH, morphoOracle, adaptiveCurveIrm, 0.86e18);
+    bytes MorphoActionsScripts = new YulHelper().getCode("MorphoScripts.sol/MorphoActions.json");
+
+    function setUp() public {
+        // Fork setup
+        vm.createSelectFork(
+            string.concat(
+                "https://node-provider.compound.finance/ethereum-mainnet/", vm.envString("NODE_PROVIDER_BYPASS_KEY")
+            ),
+            20564787 // 2024-08-19 12:34:00 PST
+        );
+        factory = new QuarkWalletProxyFactory(address(new QuarkWallet(new CodeJar(), new QuarkStateManager())));
+    }
+
+    function testRepayAndWithdrawCollateral() public {
+        vm.pauseGasMetering();
+        QuarkWallet wallet = QuarkWallet(factory.create(alice, address(0)));
+
+        deal(wstETH, address(wallet), 10e18);
+        vm.startPrank(address(wallet));
+        IERC20(wstETH).approve(morpho, 10e18);
+        IMorpho(morpho).supplyCollateral(marketParams, 10e18, address(wallet), new bytes(0));
+        IMorpho(morpho).borrow(marketParams, 1000e6, 0, address(wallet), address(wallet));
+        vm.stopPrank();
+
+        // Repay 800 USDC and withdraw 5 wstETH collateral
+        QuarkWallet.QuarkOperation memory op = new QuarkOperationHelper().newBasicOpWithCalldata(
+            wallet,
+            MorphoActionsScripts,
+            abi.encodeWithSelector(
+                MorphoActions.repayAndWithdrawCollateral.selector, morpho, marketParams, 800e6, 0, 5e18
+            ),
+            ScriptType.ScriptSource
+        );
+        (uint8 v, bytes32 r, bytes32 s) = new SignatureHelper().signOp(alicePrivateKey, wallet, op);
```

Comment:
it's confusing that this op is created here but not executed until line 80. Can we move this down so that the asertions just below immediately follow the code that actually creates the conditions?